keps/sig-apps/3973-consider-terminating-pods-deployment/README.md

```diff
@@ -0,0 +1,799 @@
+// limited by the Deployment's .spec.replicas and the DeploymentStrategy.
+Terminated DeploymentPodReplacementPolicy = "Terminated"
+)
+```
+
+```golang
+type DeploymentSpec struct {
+    ...
+    // podReplacementPolicy specifies when to create replacement Pods. 
+	// Possible values are:
+    // - Terminating policy creates replacement Pods when the old Pods start
+	//   terminating (has a .metadata.deletionTimestamp). The total number of
+	//   Deployment Pods can be greater than specified by the Deployment's
+	//   .spec.replicas and the DeploymentStrategy.
+    // - Terminated policy creates replacement Pods only when the old Pods are
+	//   fully terminated (reach Succeeded or Failed phase). The old Pods are
+	//   subsequently removed. The total number of the Deployment Pods is
+	//   limited by the Deployment's .spec.replicas and the DeploymentStrategy.
+    //
+    // The default behavior when the policy is not specified depends on the DeploymentStrategy:
+	// - Recreate strategy uses Terminated behavior when recreating the deployment,
+	//   but uses Terminating when scaling the deployment.
+	// - RollingUpdate strategy uses Terminating behavior for both rolling out and
+	//   scaling the deployments.
+	//
+	// This is an alpha field. Enable DeploymentPodReplacementPolicy to be able to
+	// use this field.
+    // +optional
+    PodReplacementPolicy *DeploymentPodReplacementPolicy `json:"podReplacementPolicy,omitempty" protobuf:"bytes,10,opt,name=podReplacementPolicy,casttype=podReplacementPolicy"`
+    ...
+}
+```
+
+```golang
+type ReplicaSetStatus struct {
+    ...
+    // Replicas is the most recently observed number of non-terminating replicas.
+    // More info: https://kubernetes.io/docs/concepts/workloads/controllers/replicationcontroller/#what-is-a-replicationcontroller
+    Replicas int32 `json:"replicas" protobuf:"varint,1,opt,name=replicas"`
+
+    // The number of non-terminating pods that have labels matching the labels of the pod template of the replicaset.
+    // +optional
+    FullyLabeledReplicas int32 `json:"fullyLabeledReplicas,omitempty" protobuf:"varint,2,opt,name=fullyLabeledReplicas"`
+
+    // readyReplicas is the number of non-terminating pods targeted by this ReplicaSet with a Ready Condition.
+    // +optional
+    ReadyReplicas int32 `json:"readyReplicas,omitempty" protobuf:"varint,4,opt,name=readyReplicas"`
+
+    // The number of available non-terminating replicas (ready for at least minReadySeconds) for this replica set.
+    // +optional
+    AvailableReplicas int32 `json:"availableReplicas,omitempty" protobuf:"varint,5,opt,name=availableReplicas"`
+
+    // The number of terminating replicas (have .metadata.deletionTimestamp) for this replica set.
+    //
+    // This is an alpha field. Enable DeploymentPodReplacementPolicy to be able to use this field.
+    // +optional
+    TerminatingReplicas int32 `json:"terminatingReplicas,omitempty" protobuf:"varint,7,opt,name=terminatingReplicas"`
```

Comment:
I'm pretty sure API rules dictate that this needs to be *int32.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#adding-unstable-features-to-stable-versions